app/src/main/java/com/xzyht/notifyrelay/sync/ProtocolTransports.kt

```
@@ -89,7 +89,37 @@ object HandshakeSender {
+            val isCharging = BatteryUtils.isCharging(manager.contextInternal)
+            val chargeSign = if (isCharging) "+" else "-"
+            val displayName = manager.encodeDisplayNameForTransportInternal(manager.localDisplayNameInternal())
+            val port = manager.listenPort
+            val payload = "HEARTBEAT_TCP:${manager.uuid}:${displayName}:${port}:${chargeSign}${batteryLevel}:android\n"
+
+            writer.write(payload)
+            writer.flush()
+            try { writer.close() } catch (_: Exception) {}
+            try { socket.close() } catch (_: Exception) {}
+            true
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)